packages/sdk-core/src/modules/storage/SessionFiles.ts

```diff
@@ -0,0 +1,244 @@
+        let session: SessionFiles | undefined = this;
+        while (session && session.sessionId !== sessionId) {
+            session = session.getPreviousSession();
+        }
+
+        return session;
+    }
+
+    public getPreviousSessions(count = Infinity) {
+        const result: SessionFiles[] = [];
+        let current = this.getPreviousSession();
+        while (current && count > 0) {
+            result.push(current);
+            count--;
+            current = current.getPreviousSession();
+        }
+
+        return result;
+    }
+
+    public lockPreviousSessions(lockId?: string) {
+        lockId = lockId ?? IdGenerator.uuid();
+        for (const session of this.getPreviousSessions(this._maxPreviousLockedSessions)) {
+            session.lock(lockId);
+        }
+        return lockId;
+    }
+
+    public unlockPreviousSessions(lockId: string) {
+        for (const session of this.getPreviousSessions(this._maxPreviousLockedSessions)) {
+            session.unlock(lockId);
+        }
+    }
+
+    public clearPreviousSessions() {
+        for (const session of this.getPreviousSessions()) {
+            session.clearSession();
+        }
+    }
+
+    public getFileName(prefix: string) {
+        this.throwIfCleared();
+
+        return (
+            this._directory +
+            '/' +
+            `${SessionFiles.escapeFileName(prefix)}_${this._escapedSessionId}_${this._timestamp}`
+        );
+    }
+
+    public getSessionFiles() {
+        this.throwIfCleared();
+
+        const files = this.readDirectoryFiles();
+        return files
+            .map((file) => SessionFiles.getFileSession(file))
+            .filter(isDefined)
+            .filter(({ sessionId }) => sessionId === this.sessionId)
+            .map(({ file }) => this._directory + '/' + file);
+    }
+
+    public clearSession() {
+        if (this._locks.size > 0) {
+            this._events.once('unlocked', () => this.clearSession());
+            return;
+        }
+
+        if (this._cleared) {
+            return;
+        }
+
+        try {
+            const sessionFiles = this.getSessionFiles();
+            for (const file of sessionFiles) {
+                this._fileSystem.unlinkSync(file);
+            }
+        } catch {
+            // Don't propagate errors
+        } finally {
+            this._cleared = true;
+        }
+    }
+
+    public lock(lockId?: string) {
+        if (this._cleared || !this._lockable) {
+            return;
+        }
+
+        lockId = lockId ?? IdGenerator.uuid();
+        this._locks.add(lockId);
+        return lockId;
+    }
+
+    public unlock(lockId: string) {
+        this._locks.delete(lockId);
+        if (this._locks.size === 0) {
+            this._events.emit('unlocked');
+        }
+    }
+
+    private static getFileSession(file: string): FileSession | undefined {
+        const [escapedSessionId, rawTimestamp] = this.splitByOneChar(file, '_').slice(-2);
```

Comment:
since you know the pattern (its defined in the getFileName) can we make it simpler to parse file name? `const [_, escapedSessionId, rawTimestamp] = file.split('_');` won't work?

Reply:
I'm taking the 2 last elements. If there are 4 elements, your example will not work.